firebase-installations/src/main/java/com/google/firebase/installations/Utils.java

```diff
@@ -31,11 +32,28 @@
   public static final long AUTH_TOKEN_EXPIRATION_BUFFER_IN_SECS = TimeUnit.HOURS.toSeconds(1);
   private static final String APP_ID_IDENTIFICATION_SUBSTRING = ":";
   private static final Pattern API_KEY_FORMAT = Pattern.compile("\\AA[\\w-]{38}\\z");
+  private static Utils singleton;
   private final Clock clock;
 
-  Utils(Clock clock) {
+  private Utils(Clock clock) {
     this.clock = clock;
   }
+
+  // Factory method that always returns the same Utils instance.
+  public static Utils getInstance() {
+    if (singleton == null) {
+      singleton = new Utils(SystemClock.getInstance());
+    }
+    return singleton;
+  }
+
+  public static Utils getTestInstance(Clock clock) {
```

Comment:
Please add JavaDoc, that the first call will define the Clock used.
If a Utils instance has already been initialized, the parameter will be ignored and the old instance will be returned.